gnovm/pkg/gnoimports/imports.go

```diff
@@ -0,0 +1,72 @@
+package gnoimports
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// PackageImports returns the list of gno imports from a given path.
```

Comment:
No call site have a MemPackage handy, I could refacto things but it seems out of scope and not really an improvement

Reply:
You are doing the exact thing that ReadMemPackage is doing here - reading a directory for .gno files. We should aim to converge code to do similar behaviour, rather than having this one to behave slightly differently from how ReadMemPackage works, for instance if we ever want to say "exclude files starting with `_`" (a rule that go has, for instance). So I would rather we refactored it now to use MemPackages; it allows us to also not have to work with a package on the filesystem (ie. in an RPC handler, for instance)

Reply:
I did it in 4908897b7e842753cf549f77920932569c6a7a4f because I feel we're bikesheding considering most of the calls will be replaced by calls to the packages loader.

To answer: `ReadMemPackage` has a different behavior and more responsibilities than just filtering files. In fact I have to wrap `ReadMemPackage` in a recovering `defer` [here](https://github.com/gnolang/gno/pull/3123/files#diff-f3a4b19eeed17c3197d9eb7ae51c55edfd50f8e18b5048de314e840b283bcb85R102) to adjust to the behavior change.

Reply:
Actually I would like to revert https://github.com/gnolang/gno/commit/4908897b7e842753cf549f77920932569c6a7a4f. `ReadMemPackage` is very strict and not suited for imports analysis IMO.

Reply:
Okay I changed the `ReadMemPackage` api in 9e62b83edbd15535c6409ef488f16cc8f70fa0f3 and now it's okay to keep https://github.com/gnolang/gno/commit/4908897b7e842753cf549f77920932569c6a7a4f IMO